### PR TITLE
chore: add hot reload for extension background

### PR DESCRIPTION
**Pull request by IronLu233**
# Motivation
Take auto refresh to the chrome extension on development.When background script or content script changes, the plugin will automatically reload.
# Modifications
- split develop build into two parts: frontend(popup js and it's HTML) and background(background service worker, content script)
- start a websocket server when run `npm start`
- inject a script for create ws client on background service worker. That can react bundle change, and reload Chrome extension.
- move some directories. Background and ContentScript should not in page direcotry.
- keep the production build as same

### CODE REVIEW

**Review by zhangyouxin — APPROVED** [review]

File: packages/extension-chrome/package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
   },
   "devDependencies": {
     "@pmmmwh/react-refresh-webpack-plugin": "0.5.10",
+    "@types/rimraf": "^3.0.2",
```
Comment (homura) [inline packages/extension-chrome/package.json:38]:
```suggestion
    "@types/rimraf": "3.0.2",
```

We'll use Renovate to upgrade dependencies, so we can make the dependency versions to be fixed

**Review by zhb666 — APPROVED** [review]

File: packages/extension-chrome/package.json [hunk 2]
```diff
@@ -43,13 +44,17 @@
...
     "terser-webpack-plugin": "5.3.6",
     "ts-loader": "9.4.2",
+    "ts-node": "^10.9.1",
     "webpack": "5.75.0",
     "webpack-cli": "5.0.1",
     "webpack-dev-server": "4.11.1",
-    "webpack-merge": "5.8.0"
+    "webpack-inject-plugin": "1.5.5",
+    "webpack-merge": "5.8.0",
+    "ws": "^8.11.0"
```
Comment (homura) [inline packages/extension-chrome/package.json:58]:
```suggestion
    "ws": "8.11.0"
```

File: packages/extension-chrome/package.json [hunk 3]
```diff
@@ -43,13 +44,17 @@
     "fork-ts-checker-webpack-plugin": "7.2.13",
     "html-loader": "4.2.0",
     "html-webpack-plugin": "5.5.0",
+    "rimraf": "3.0.2",
     "source-map-loader": "4.0.1",
     "style-loader": "3.3.1",
     "terser-webpack-plugin": "5.3.6",
     "ts-loader": "9.4.2",
+    "ts-node": "^10.9.1",
```
Comment (homura) [inline packages/extension-chrome/package.json:52]:
```suggestion
    "ts-node": "10.9.1",
```

**Review by homura — APPROVED** [review]
LGTM

**Pull request merged**